src/javascript/app_2/Modules/Trading/Containers/purchase.jsx

```diff
@@ -57,24 +60,22 @@ const Purchase = ({
                 onMouseEnter={() => { onHoverPurchase(true, type); }}
                 onMouseLeave={() => { onHoverPurchase(false); }}
             >
-                {is_purchase_error &&
-                <MessageBox
-                    purchase_info={purchase_info}
-                    onClick={resetPurchase}
-                    currency={currency}
-                />
-                }
                 {(is_purchase_locked && idx === 0) &&
                 <PurchaseLock onClick={togglePurchaseLock} />
                 }
                 <React.Fragment>
                     <ContractInfo
+                        basis={basis}
                         currency={currency}
                         proposal_info={info}
                         has_increased={info.has_increased}
+                        is_loading={is_loading}
                         is_visible={!is_contract_mode}
                     />
-                    <div className='btn-purchase__shadow-wrapper'>
+                    <div className={classNames('btn-purchase__shadow-wrapper', { 'btn-purchase__shadow-wrapper--disabled': (is_proposal_error || is_disabled) })}>
+                        {is_proposal_error &&
```

Comment:
space and align tooltip ?

Reply:
it's following the same style as line 67-69